internal/store.go

```diff
@@ -0,0 +1,267 @@
+package internal
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/google/uuid"
+	"github.com/hbomb79/Thea/internal/database"
+	"github.com/hbomb79/Thea/internal/ffmpeg"
+	"github.com/hbomb79/Thea/internal/media"
+	"github.com/hbomb79/Thea/internal/transcode"
+	"github.com/hbomb79/Thea/internal/workflow"
+	"github.com/hbomb79/Thea/internal/workflow/match"
+	"github.com/jmoiron/sqlx"
+	"github.com/lib/pq"
+)
+
+const (
+	PgFkConstraintViolationCode = "23503"
+)
+
+var (
+	ErrWorkflowTargetIDMissing = errors.New("one or more of the targets provided cannot be found")
+)
+
+type (
+	// storeOrchestrator is responsible for managing all of Thea's resources,
+	// especially highly-relational data. You can think of all
+	// the data stores below this layer being 'dumb', and this store
+	// linking them together and providing the database instance
+	//
+	// If consumers need to be able to access data stores directly, they're
+	// welcome to do so - however caution should be taken as stores have no
+	// obligation to take care of relational data (which is the orchestrator's job)
+	storeOrchestrator struct {
+		db             database.Manager
+		MediaStore     *media.Store
+		TranscodeStore *transcode.Store
+		WorkflowStore  *workflow.Store
+		TargetStore    *ffmpeg.Store
```

Comment:
Even if this struct were public, you probably wouldn't want the internal store fields to be public. Having them exported would imply to me that you must be doing something funny with them somewhere... but it doesn't look like you are.

Reply:
Ah yup, good point. I think initially I wanted to allow for consumers of the orchestrator to grab them and use them directly, but I kinda changed my mind half way through ;)